subprojects/jvm-services/src/main/java/org/gradle/jvm/toolchain/internal/MavenToolchainsInstallationSupplier.java

```diff
@@ -85,7 +88,22 @@ protected Set<InstallationLocation> findCandidates() {
                 for (int i = 0; i < nodes.getLength(); i++) {
                     Node item = nodes.item(i);
                     if (item != null && item.getNodeType() == Node.TEXT_NODE) {
-                        locations.add(item.getNodeValue().trim());
+                        String nodeValue = item.getNodeValue().trim();
+                        Matcher matcher = ENV_VARIABLE_PATTERN.matcher(nodeValue);
+                        StringBuffer resolvedValue = new StringBuffer();
+                        if (matcher.find()) {
```

Comment:
🤔 Your setup for doing the replacement sounds sane. But I can imagine a value that would be composed of multiple env vars.

Could you look into supporting something like:
`<jdkHome>${env.JDKHOMES}/${env.ZULU_17}</jdkHome>`

Reply:
Thank you for pointing that out. Its now made in last changes